### PR TITLE
Upgrade Ember to 4.12

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

- Upgrade Ember to the latest stable released version (4.12) using `ember-cli-update`
- Upgrade minor and patch dependencies using `yarn upgrade-interactive`
- Align our supported version of Node.js with Ember, shifting from `12.* || 14.* || >= 16` to `14.* || 16.* || >= 18`

Note `ember-template-lint` 5.x is a breaking release that includes linting Handlebar syntax in test files, affecting the way we indent template calls in all our tests. For this reason, I suggest we leave this update outside the scope of this upgrade.

### :hammer_and_wrench: Detailed description

Did this while drafting [DS-050](https://go.hashi.co/rfc/ds-050) and [DS-051](https://go.hashi.co/rfc/ds-050)

### :link: External links

Jira ticket: [HDS-1925](https://hashicorp.atlassian.net/browse/HDS-1925)

***

### 👀 Reviewer's checklist:

- [x] +1 Percy if applicable
- [x] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1925]: https://hashicorp.atlassian.net/browse/HDS-1925?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by alex-ju** [comment]
Sadly no difference 😞

File: website/package.json [hunk 1]
```diff
@@ -68,18 +68,18 @@
...
     "ember-meta": "^2.0.0",
     "ember-page-title": "^7.0.0",
-    "ember-power-select": "^6.0.0",
+    "ember-power-select": "^6.0.1",
     "ember-prism": "^0.13.0",
     "ember-qunit": "^6.2.0",
     "ember-resolver": "^8.1.0",
     "ember-router-scroll": "^4.1.2",
-    "ember-source": "~4.8.4",
+    "ember-source": "~4.8.5",
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply website/package.json:76]: sure, can do that. I focused on the packages in this PR, but can bundle the website too

[the remark this answers is not in the document]
Comment (Dhaulagiri) [reply website/package.json:76]: Don't let me overcomplicate things. I was more curious.

**Comment by Dhaulagiri** [comment]
> Note ember-template-lint 5.x is a breaking release that includes linting Handlebar syntax in test files, affecting the way we ident template calls in all our tests. For this reason, I suggest we leave this update outside the scope of this upgrade.

👍  doing this in a separate PR seems prudent (probably after we land all the in-flight work we have at the moment)

**Review by Dhaulagiri — APPROVED** [review]
Left one suggestion and one comment to myself but overall I think this looks good to me.

File: packages/ember-flight-icons/config/ember-try.js [hunk 2]
```diff
@@ -19,18 +19,18 @@ module.exports = async function () {
     },
     scenarios: [
       {
-        name: 'ember-lts-3.24',
+        name: 'ember-lts-3.28',
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply packages/ember-flight-icons/config/ember-try.js:22]: yes, I noticed that – could leave a comment on the file maybe, or comment the scenarios, to make it clear that they don't run

[the remark this answers is not in the document]
Comment (alex-ju) [reply packages/ember-flight-icons/config/ember-try.js:22]: yeah, ideally we'd merge them 😬  happy to start an RFC for this

**Comment by MelSumner** [comment]
Dropping Node version support is considered a breaking change anyway, so maybe it's fine to add other breaking changes here.

**Comment by alex-ju** [comment]
> Dropping Node version support is considered a breaking change anyway, so maybe it's fine to add other breaking changes here.

Considering we never stated our Node.js support until now, and we didn't/don't have any consumers on v12 according to their manifest, I thought this is the first time we'd made a conscious decision about the supported version of Node.js. Based on this consideration I went with a patch, but happy to discuss this further.

**Pull request merged**